### ISSUE TITLE
[BUG] [Linter] T038 does not handle Jinja whitespace-control tag variants

### ISSUE THREAD

**Issue by anentropic** [djlint/djLint#2396]
- [x] I'm on the [latest version](https://pypi.org/project/djlint/) of djLint
- [x] I've searched the [issues](https://github.com/djlint/djLint/issues)
- [x] I've read the [docs](https://djlint.com)

## System Info

- OS: macOS 14.6.1
- Python Version: 3.11
- djLint Version: 1.44.2
- template language: Jinja2

## Issue

`T038` Block tag has no matching end tag.

Is marked as "applies generally to templates" but there are some Jinja cases which cause false positive

Specifically the `+` form of https://jinja.palletsprojects.com/en/stable/templates/#whitespace-control

In these cases Jinja2 allows 'mismatched' tags because the start and/or end tag may have modifying markers like:

```jinja
{% for item in seq +%}
    {{ item }}
{%+ endfor %}
```

## How To Reproduce

Apply djlint to Jinja templates using whitespace control

**Comment by ifm-pgarner** [comment]
Turn out this is specific to the `{%+ ... %}` version of the tags which disable block trimming behaviour, `{%- ... %}` is ok
